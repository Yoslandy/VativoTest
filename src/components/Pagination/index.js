import React from "react"
// import * as converter from "hex2dec"
import PropTypes from "prop-types"
import Button from "../Button"
import Icon from "../Icon"

export default function Pagination(props) {
    const { dataPagination, className, handleLastPage, handlePage } = props

    // const decimal = converter.hexToDec("0xFA")

    // const page = ["MQ==", "Ng=="].includes(datePagination.startCursor)
    //     ? "1"
    //     : converter.hexToDec(datePagination.startCursor)
    // const lastPage = ["MQ==", "Ng=="].includes(datePagination.endCursor)
    //     ? "1"
    //     : converter.hexToDec(datePagination.endCursor)

    return (
        <div className={`w-[255px] grid grid-cols-3 items-center ${className}`}>
            <span className="text-black font-roboto text-sm font-medium">
                {1} of {10}
            </span>
            <Button
                type=""
                className="!bg-white text-primary-60 w-[80px] h-[25px] font-roboto"
                text="Next"
                iconRigth={<Icon type="arrow" className="ml-2" />}
                onClick={handlePage}
                disabled={!dataPagination.hasNextPage}
            />
            <Button
                type=""
                className="!bg-white text-primary-60 w-[80px] h-[25px] font-roboto"
                text="Last"
                iconRigth={<Icon type="double_arrow" className="ml-2" />}
                onClick={handleLastPage}
                disabled={!dataPagination.hasPreviousPage}
            />
        </div>
    )
}

Pagination.propTypes = {
    dataPagination: PropTypes.shape({
        startCursor: PropTypes.string,
        endCursor: PropTypes.string,
        hasNextPage: PropTypes.bool,
        hasPreviousPage: PropTypes.bool,
    }),
    handlePage: PropTypes.func,
    handleLastPage: PropTypes.func,
    className: PropTypes.string,
}

Pagination.defaultProps = {
    dataPagination: {
        startCursor: "1",
        endCursor: "1",
        hasNextPage: true,
        hasPreviousPage: false,
    },
    className: "",
    handlePage: () => {},
    handleLastPage: () => {},
}
