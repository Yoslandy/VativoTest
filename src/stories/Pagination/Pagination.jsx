import React from "react"
import PropTypes from "prop-types"

import Pagination from "../../components/Pagination"

export default function PaginationCustom({ page, lastPage }) {
    return <Pagination page={page} lastPage={lastPage} />
}

PaginationCustom.propTypes = {
    page: PropTypes.number.isRequired,
    lastPage: PropTypes.number.isRequired,
}
