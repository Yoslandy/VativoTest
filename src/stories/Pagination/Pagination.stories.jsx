import React from "react"
import "../../assets/styles/build.css"

import Pagination from "./Pagination"

export default {
    title: "Example/Pagination",
    component: Pagination,
    argTypes: {
        page: 1,
        lastPage: 10,
    },
}

function Template(args) {
    return <Pagination {...args} />
}

export const Default = Template.bind({})
Default.args = {
    page: 1,
    lastPage: 10,
}
