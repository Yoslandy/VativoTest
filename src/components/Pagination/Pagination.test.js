import React from "react"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import Pagination from "./index"

test("Renders Pagination form correctly", () => {
    const component = renderer.create(
        <MemoryRouter>
            <Pagination lastPage={10} page={1} />
        </MemoryRouter>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
