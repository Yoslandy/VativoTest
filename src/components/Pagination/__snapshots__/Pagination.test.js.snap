// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Pagination form correctly 1`] = `
<div
  className="w-[255px] flex justify-between items-center "
>
  <span
    className="text-black font-roboto text-sm font-medium"
  >
    1
     of 
    10
  </span>
  <button
    className="button undefined h-8 w-[100px] bg-primary-100 !bg-white text-primary-60 w-[80px] h-[25px]"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
     
    Next
     
    <span
      className="material-icons ml-2"
      onClick={[Function]}
      role="button"
    >
      keyboard_arrow_right
    </span>
  </button>
  <button
    className="button undefined h-8 w-[100px] bg-primary-100 !bg-white text-primary-60 w-[80px] h-[25px]"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
     
    Last
     
    <span
      className="material-icons ml-2"
      onClick={[Function]}
      role="button"
    >
      keyboard_double_arrow_right
    </span>
  </button>
</div>
`;
